content/docs/development/health-rules.md

```diff
@@ -0,0 +1,168 @@
+      type: Ready
+    - status: "True"
+      type: BuilderReady
+```
+
+Given that it makes use of the `status.conditions` pattern (and `Ready: True` indicates healthiness), we can leverage
+`singleConditionType`:
+
+<!-- prettier-ignore-start -->
+```yaml
+  healthRule:
+    singleConditionType: Ready
+```
+<!-- prettier-ignore-end -->
+
+Cartographer will then consider the named condition (in this case `Ready`) and evaluate healthiness as the status on
+that condition:
+
+- healthy: status == true
+- unhealthy: status == false
+- unknown: anything else
+
+The `message` field of the specified condition will be replicated on the owner object.
+
+## Multi Match
+
+With multiMatch we're able to specify more than one matching rule for determining healthiness. This is the most flexible
+of the health rules.
+
+For some controllers, two conditions must be met for the object to be healthy. For example in Deployments, users want
+both the `Available` and `Progressing` conditions to be true. For other resources, one condition indicates the object is
+healthy and another condition indicates that the object is unhealthy. Kapp's `App` resource behaves in this manner, if
+the `ReconcileSucceeded` condition is true the object is healthy, while if the `ReconcileFailed` condition is true the
+object is unhealthy. Multimatch can address both of these use cases.
+
+When specifying multiMath, users must define both what constitutes healthy and what indicates unhealthy. Users may
+specify multiple matchers. The matchers for healthy must all be met for an object to be healthy. If any of the matchers
+under unhealthy are met, the object is considered unhealthy.
+
+<!-- prettier-ignore-start -->
+```yaml
+  healthRule:
+    multiMatch:
+      healthy: #! matchers here are ANDed
+        ...
+      unhealthy: #! matchers here are ORed
+        ...
+```
+<!-- prettier-ignore-end -->
+
+There are two types of matchers available, matchConditions and matchFields.
+
+### Match Conditions
+
+MultiMatch's MatchConditions provide more nuance to SingleCondition. Users specify the `type` of the condition on the
+object that should be inspected as well as the `status` value which is considered a match. When a matcher set is
+satisfied, the `message` field of the first condition will be replicated on the owner object.
+
+As an example, we can replicate the behavior of the single condition type that we observed above.
+
+<!-- prettier-ignore-start -->
+```yaml
+  healthRule:
+    multiMatch:
+      healthy:
+        - matchConditions:
+            type: Ready
+            status: True
+      unhealthy:
+        - matchConditions:
+            type: Ready
+            status: False
+```
+<!-- prettier-ignore-end -->
+
+### Match Fields
+
+Match fields allow users to inspect arbitrary fields on an object in order to determine health. This is useful for
+objects which do not use conditions. Match fields also allow users to specify arbitrary fields on the object which
```

Comment:
another good use for `matchFields` despite an object making use of `status.conditions` is `messagePath` - that's the only matcher that lets you tell cartographer where to look for a friendly message, thus, very useful for those like `kapp-controller/App`:

```yaml
  healthRule:
    multiMatch:
      healthy:
        matchConditions:
          - type: ReconcileSucceeded
            status: 'True'
      unhealthy:
        matchFields:
          - key: 'status.conditions[?(@.type=="ReconcileFailed")].status'
            operator: 'In'
            values: ['True']
            messagePath: '.status.usefulErrorMessage'        # `kapp-controller` puts a nice message in this field,
                                                             # not the condition.message
```

Reply:
That's what I was trying to explain with this sentence
> Match fields also allow users to specify arbitrary fields on the object which explain the reason for failure.

I've tried some additional wording to clarify:

> Match fields also allow users to specify arbitrary fields on the object for messagePath, which explains the reason for failure to an end user.